tools/construct_npy.cpp

```diff
@@ -58,25 +60,40 @@ void buildIndex(float *data,
 }
 
 void run(float *data, flatnav::METRIC_TYPE metric_type, int N, int M, int dim,
-         int ef_construction, const std::string &save_file,
-         bool quantize = false) {
+         int ef_construction, const std::string &save_file, int quantize) {
 
   if (quantize) {
-    // Parameters M and nbits should be adjusted accordingly.
-    auto quantizer = std::make_shared<ProductQuantizer>(
-        /* dim = */ dim, /* M = */ 8, /* nbits = */ 8,
-        /* metric_type = */ metric_type);
-
-    auto start = std::chrono::high_resolution_clock::now();
-    quantizer->train(/* vectors = */ data, /* num_vectors = */ N);
-    auto stop = std::chrono::high_resolution_clock::now();
-    auto duration =
-        std::chrono::duration_cast<std::chrono::milliseconds>(stop - start);
-    std::clog << "Quantization time: " << (float)duration.count()
-              << " milliseconds" << std::endl;
-
-    buildIndex<ProductQuantizer>(data, std::move(quantizer), N, M, dim,
-                                 ef_construction, save_file);
+
+    if (quantize == 1) {
```

Comment:
Can we make the quantizer choice be a public member enum for the flatnav class? similar to how we did the reordering options? This will be cleaner.